### PR TITLE
fix(www): update docs regarding client initialization

### LINKED ISSUE #2730: docs not fully updated to reflect changes on client initialization
### Provide environment information

  System:
    OS: macOS 12.5.1
    CPU: (8) arm64 Apple M1
    Memory: 132.70 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.15.1/bin/yarn
    npm: 8.18.0 - ~/.nvm/versions/node/v16.15.1/bin/npm
  Browsers:
    Brave Browser: 100.1.37.113
    Chrome: 105.0.5195.125
    Firefox Developer Edition: 104.0
    Safari: 15.6.1
  npmPackages:
    @trpc/client: 10.0.0-proxy-alpha.77 => 10.0.0-proxy-alpha.77 
    @trpc/next: 10.0.0-proxy-alpha.77 => 10.0.0-proxy-alpha.77 
    @trpc/react: 10.0.0-proxy-alpha.77 => 10.0.0-proxy-alpha.77 
    @trpc/server: 10.0.0-proxy-alpha.77 => 10.0.0-proxy-alpha.77 
    next: 12.2.5 => 12.2.5 
    react: 18.2.0 => 18.2.0

### Describe the bug

With version 10.0.0-proxy-alpha.77 and newer the types of the config field in `createTRPCNext` from `@trpc/next` are incorrect. I'm not sure what is the issue, or what changed, but I'll provide a reproduction.

<details>
<summary>Photo of type error</summary>

image

</details>

### To reproduce

[stackblitz](https://stackblitz.com/edit/trpc-trpc-r5q6yb?file=src/utils/trpc.ts) forked from the homepage with one change.

I changed the config to match what is listed in the [documentation](https://trpc.io/docs/v10/nextjs#4-create-trpc-hooks). In the documentation, `url` is returned in the config rather than `links`. Should `links` be used instead? If so, let's update the documentation!

```diff
export const trpc = createTRPCNext<AppRouter>({
  config() {
    return {
+      url: getBaseUrl() + '/api/trpc',
-       links: [
-        httpBatchLink({
-          url: getBaseUrl() + '/api/trpc',
-        }),
-      ],
    };
  },
  ssr: true,
});
```

### Additional information

Thanks for the help! Please let me know if you need more information to debug this issue.

If you're already aware of the type regression, please close this. Thanks :)

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### PR DESCRIPTION
**Pull request by charkour**
Closes #2730

## 🎯 Changes

Update various documentation to use current API for client initialization.

Thanks for this great library! Please make any changes to this PR as desired. Thank you

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/docs/client/cors.md [hunk 1]
```diff
@@ -10,15 +10,19 @@ If your API reside on a different origin than your front-end and wish to send co
 The arguments provided to the fetch function used by tRPC can be modified as follow.
 
 ```ts title='app.ts'
-import { createTRPCClient } from '@trpc/client';
+import { createTRPCProxyClient, httpBatchLink } from '@trpc/client';
```
Comment (charkour) [inline www/docs/client/cors.md:13]: Was not sure if `httpBatchLink` is something the docs should default to

Comment (charkour) [reply www/docs/client/cors.md:13]: Also, in this specific case, I noticed that `createTRPCClient` was deprecated for `createTRPCProxyClient` so I made that change

File: www/docs/migration/migrate-from-v9-to-v10.mdx [hunk 2]
```diff
@@ -357,7 +357,7 @@ const client = createTRPCClient({
 });
 
 // after:
-import { createTRPCProxyClient, httpLink } from '@trpc/client';
+import { createTRPCProxyClient, httpBatchLink } from '@trpc/client';
```
Comment (charkour) [inline www/docs/migration/migrate-from-v9-to-v10.mdx:360]: Fix typo

**Review by KATT — APPROVED** [review]

File: www/docs/nextjs/ssr.md [hunk 3]
```diff
@@ -19,7 +21,12 @@ export const trpc = createTRPCNext<AppRouter>({
       // during client requests
       return {
         transformer: superjson, // optional - adds superjson serialization
-        url: '/api/trpc',
+        links: [
+          // optional - adds batching
```
Comment (juliusmarminge) [inline www/docs/nextjs/ssr.md:25]: Bit missleading

File: www/docs/server/caching.md [hunk 4]
```diff
@@ -22,14 +22,20 @@ If you turn on SSR in your app you might discover that your app loads slow on fo
...
 import { createTRPCNext } from '@trpc/next';
 import { AppRouter } from '../server/routers/_app';
 
 export const trpc = createTRPCNext<AppRouter>({
   config({ ctx }) {
     if (typeof window !== 'undefined') {
       return {
-        url: '/api/trpc',
+        links: [
+          // optional - adds batching
```
Comment (juliusmarminge) [inline www/docs/server/caching.md:34]: ^^

File: www/docs/server/caching.md [hunk 5]
```diff
@@ -38,7 +44,12 @@ export const trpc = createTRPCNext<AppRouter>({
       : 'http://localhost:3000/api/trpc';
 
     return {
-      url,
+      links: {
+        // optional - adds batching
```
Comment (juliusmarminge) [inline www/docs/server/caching.md:48]: ^^

File: www/docs/nextjs/ssr.md (hunk 3 above)
Comment (KATT) [reply www/docs/nextjs/ssr.md:25]:
```suggestion
```

File: www/docs/server/caching.md (hunk 5 above)
Comment (KATT) [reply www/docs/server/caching.md:48]:
```suggestion
```

File: www/docs/server/caching.md (hunk 4 above)
Comment (KATT) [reply www/docs/server/caching.md:34]:
```suggestion
```

**Pull request merged**